ai/executors/pathfinder_module.py

```diff
@@ -1,42 +1,52 @@
 
-
-from ai.Algorithm.path_partitionner import PathPartitionner, CollisionBody
+import logging
+from collections import defaultdict
+from typing import Dict
+from Util import Singleton
+from ai.Algorithm.path_partitionner import PathPartitionner
 
 MIN_DISTANCE_FROM_OBSTACLE = 250
 
 
-def create_pathfinder():  # FIXME: we should not create a new pathfinder object each time...
-
-    return PathPartitionner()
-
-
-def generate_path(game_state, player, ai_command):
-
-    pathfinder = create_pathfinder()
-
-    collision_bodies = get_pertinent_collision_objects(player, game_state, ai_command)
+class PathfinderModule(metaclass=Singleton):
```

Comment:
Since only PlayExecutor use PathfinderModule, I don't think it need to be a singleton.

Reply:
oups my bad, copy-paste